diff --git a/dist/NodeFileSystem.js b/dist/NodeFileSystem.js
index a94f50e03afa1777453ca4f04508e85fa241a1b8..597a67648a507ceb1120b61ae850bbdc49f05ad1 100644
--- a/dist/NodeFileSystem.js
+++ b/dist/NodeFileSystem.js
@@ -232,7 +232,7 @@ const makeFile = /*#__PURE__*/(() => {
     write(buffer) {
       return Effect.suspend(() => {
         const position = this.position;
-        return Effect.flatMap(this.append ? Effect.succeed(undefined) : positionToNumber(position, "write"), nodePosition => Effect.map(nodeWrite(this.fd, buffer, undefined, undefined, nodePosition), bytesWritten => {
+        return Effect.flatMap(this.append ? Effect.succeed(undefined) : positionToNumber(position, "write"), nodePosition => Effect.map(nodeWrite(this.fd, buffer, 0, buffer.length, nodePosition), bytesWritten => {
           if (!this.append) {
             this.position = position + BigInt(bytesWritten);
           }
@@ -243,7 +243,7 @@ const makeFile = /*#__PURE__*/(() => {
     writeAllChunk(buffer) {
       return Effect.suspend(() => {
         const position = this.position;
-        return Effect.flatMap(this.append ? Effect.succeed(undefined) : positionToNumber(position, "writeAll"), nodePosition => Effect.flatMap(nodeWriteAll(this.fd, buffer, undefined, undefined, nodePosition), bytesWritten => {
+        return Effect.flatMap(this.append ? Effect.succeed(undefined) : positionToNumber(position, "writeAll"), nodePosition => Effect.flatMap(nodeWriteAll(this.fd, buffer, 0, buffer.length, nodePosition), bytesWritten => {
           if (bytesWritten === 0) {
             return Effect.fail(Error.systemError({
               module: "FileSystem",
